src/app/views/credentials.go

```diff
@@ -224,6 +226,170 @@ func credentialsGroup(router *gin.Engine) {
 		c.JSON(http.StatusCreated, cv)
 	})
 
+	g.POST("/import", auth.LoginRequired(), func(c *gin.Context) {
+
+		u, _ := c.Get("user")
+		user := u.(*models.User)
+
+		file, _, err := c.Request.FormFile("file")
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{
+				"error":   err.Error(),
+				"message": "Couldn't upload media",
+			})
+			return
+		}
+		defer file.Close()
+
+		SchemaID := c.Request.FormValue("schema_id")
+
+		// Fetching Schema
+		schema, err := models.GetSchema(uuid.MustParse(SchemaID))
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
+		if schema.IssueDisabled {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "schema for issuing credentials is disabled"})
+			return
+		}
+
+		schemaAttributes := map[string]string{}
+		for _, attributes := range schema.Attributes {
+			schemaAttributes[attributes.Name] = string(attributes.Type)
+		}
+		schemaAttributes["first_name"] = string(models.Text)
+		schemaAttributes["last_name"] = string(models.Text)
+		schemaAttributes["email"] = string(models.Email)
+
+		//Processing CSV file
+		resultChan, errChan := make(chan bool), make(chan error)
+
+		go lib.ValidateCSVFile(file, schemaAttributes, resultChan, errChan)
+
+		for {
+			select {
+			case _ = <-resultChan:
+				c.JSON(http.StatusCreated, gin.H{"message": "success"})
+				go services.InitiateCSVImport(file, map[string]any{
```

Comment:
you process the io reader 2 time you can make it 1 time and push data to ram then pass data to InitateCSVImport instead of seek file to 0, 0